main/windows/index.ts

```diff
@@ -609,5 +609,8 @@ export default {
   min(e: IpcMainEvent) {
     windowFromWebContents(e.sender).minimize()
   },
+  browserWindows() {
```

Comment:
exposing this feels like a pretty big API change for this module, is there a specific reason why we need to? this feels like maybe something we should be able to access via the state

Reply:
Putting a whole BrowserWindow object on the state isn't going to work, open to other ideas though.  Maybe we delegate the dialog handling to the window index.

Reply:
Still doesn't feel quite right.  Maybe move the whole thing into the window index.